### PR TITLE
Fix-role-management-dp-91

### PR DESCRIPTION
**Pull request by ericmaniraguha**
# PR Description

This PR is for fixing role management, display all users based on their role. 
Note: This is the Desktop Version.

# Description of tasks that were expected to be completed

- [x] Changing the way code was written, date was hardcoded.

- [x] The UI The way data was displayed on the card
 
# How has this been tested?

1. Clone the repository by running git clone [Link](https://github.com/atlp-rwanda/atlp-pulse-fn.git)
2. run npm install
3. checkout branch **fix-role-management-dp-91**
4. npm run dev

or

[Link](https://devpulse-review-app-54.herokuapp.com/dashboard/manage)
login as admin{
email: admin@gmail.com
password: test123
}

# Screenshots (If appropriate)
Screenshot 2022-08-07 120701

### CODE REVIEW

**Comment by feyton** [comment]
I deployed a review app for you to try out this pull request 👇🏽

Heroku App: https://devpulse-review-app-54.herokuapp.com/


<hidden purpose="for-rewritable-pr-comment-action-use" value="Deploy"></hidden>

**Comment by ericmaniraguha** [comment]
> 

Thank you @feyton

**Review by chanceDira — CHANGES_REQUESTED** [review]
Good work @ericmaniraguha,

- Can you fix some typos. like in the table row the role can be(Admin, Coordinator, Trainee).
- The table navbar we can have Coordinators instead of Coordinator.
- You can remove scroll for this model.

Screen Shot 2022-08-09 at 11 26 31

Screen Shot 2022-08-09 at 11 30 33

File: src/containers/admin-dashBoard/tests/ManageRole.test.tsx [hunk 1]
```diff
@@ -16,9 +16,9 @@ describe('Manage role page tests', () => {
         <ManageRole />
       </BrowserRouter>,
     );
-    let removeManageRole = getByTestId('remove');
-    fireEvent.click(removeManageRole);
-    expect(removeRoleModel).toHaveBeenCalledTimes(0);
+    // let removeManageRole = getByTestId('remove');
+    // fireEvent.click(removeManageRole);
+    // expect(removeRoleModel).toHaveBeenCalledTimes(0);
```
Comment (feyton) [inline src/containers/admin-dashBoard/tests/ManageRole.test.tsx:21]: Since this codeblock is not needed, you can remove it.

Instead of commenting out the codeblock if you have added changes that removes them.

File: src/dummyData/rolemanagement.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
+[
+    {
+      "name": "Keza Alice",
+      "email":"keza@gmail.com",
+      "status": "active",
+      "role": "admin" ,
+      "value":"1"
```
Comment (feyton) [inline src/dummyData/rolemanagement.json:7]: What is the use of the value attribute in the object.

It may not be related to the PR but if this value is not related to the user model, you can safely remove it.

File: src/containers/admin-dashBoard/ManagerRoles.tsx [hunk 3]
```diff
@@ -211,13 +222,17 @@ const AdminSission = () => {
                       </tr>
                     </thead>
                     <tbody>
-                      <tr>
-                        <td className="border-b border-gray-200 bg-white dark:bg-dark-bg">
-                          <div className="flex items-left">
-                            <div className="md:grid md:grid-cols-2">
-                              <p></p>
+                      {rolemanagement.map((role: any, index: any) => {
```
Comment (feyton) [inline src/containers/admin-dashBoard/ManagerRoles.tsx:225]: Is it a role or a user. Please change the naming to ensure that everyone is aware of the object that is being destructured.

We can not expect to have a role that has a name attribute

**Review by feyton — CHANGES_REQUESTED** [review]
@ericmaniraguha 
This is very great work. Please look into the comments I left. Everything else is improved.

**Review by jeandamascyifuzo — APPROVED** [review]
well done

**Review by IIshimwe — APPROVED** [review]

**Review by chanceDira — APPROVED** [review]
Nice job @jeandamascyifuzo @ericmaniraguha

**Review by ceelogre — COMMENTED** [review]
- Is there a pagination implementation?
- Does filtering work (admin tab, trainees,..)? It doesn't seem to work when I try it
- How comes when I click on `remove` then select cancel in the new window, it reloads the page?
- Search feature doesn't seem to work too

Let me know what approach you took took to implementing this feature.

**Review by ceelogre — APPROVED** [review]

**Review by feyton — APPROVED** [review]
